spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/batch/AbstractSimpleJobServiceTests.java

```diff
@@ -192,6 +194,18 @@ void exceptionsShouldBeThrownIfRequestForNonExistingJobInstance() {
 	@Test
 	void stoppingJobExecutionShouldLeaveJobExecutionWithStatusOfStopping() throws Exception {
 		JobExecution jobExecution = createJobExecution(BASE_JOB_INST_NAME,true);
+		assertJobHasStopped(jobExecution);
+	}
+
+	@Test
+	void stoppingAllJobExecutionsShouldLeaveJobExecutionsWithStatusOfStopping() throws Exception {
+		JobExecution jobExecutionOne = createJobExecution(BASE_JOB_INST_NAME,true);
+		JobExecution jobExecutionTwo = createJobExecution(BASE_JOB_INST_NAME+"_TWO",true);
+		assertJobHasStopped(jobExecutionOne);
+		assertJobHasStopped(jobExecutionTwo);
+	}
+
+	private void assertJobHasStopped(JobExecution jobExecution) throws NoSuchJobExecutionException, JobExecutionNotRunningException {
 		jobExecution = jobService.getJobExecution(jobExecution.getId());
 		assertThat(jobExecution.isRunning()).isTrue();
```

Comment:
Should this be `isFalse()` ?

Reply:
In this case it should be running because the JobExecution is created in a running state.   This test verifies the pre-condition that the test running before we call the stop on line 212.

Reply:
Ahh yeh, I did not jump into `assertJobHasStopped` as the name implies that it is only asserting that the job has stopped. 

I think we can simplify this a bit by moving the post-creation verify steps
```java
jobExecution = jobService.getJobExecution(jobExecution.getId());
assertThat(jobExecution.isRunning()).isTrue();
assertThat(jobExecution.getStatus()).isNotEqualTo(BatchStatus.STOPPING);
```
into the `createJobExecution` method.

Then we can move the action (`stop`) into the test followed by a simplified assert method. That way it would read more like `AAA` (arrange / act / assert).

Something like this:
```java
@Test
void stoppingJobExecutionShouldLeaveJobExecutionWithStatusOfStopping() throws Exception {
	JobExecution jobExecution = createJobExecution(BASE_JOB_INST_NAME,true);
	jobService.stop(jobExecution.getId());
	assertJobHasStopped(jobExecution);
}

private void assertJobHasStopped(JobExecution jobExecution) throws NoSuchJobExecutionException, JobExecutionNotRunningException {
	jobExecution = jobService.getJobExecution(jobExecution.getId());
	assertThat(jobExecution.getStatus()).isEqualTo(BatchStatus.STOPPING);
	assertThat(jobExecution.isRunning()).isTrue();
}
```

Wdyt?